zkir/Utils/ParserUtils.cpp

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2025 The ZKIR Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "zkir/Utils/ParserUtils.h"
+
+namespace mlir::zkir {
+
+namespace {
+
+ParseResult doParsePrimeField(OpAsmParser &parser, APInt &parsedInt,
```

Comment:
can we rename to `doParseModulus`?

Reply:
Actually this is parsing prime field or modular integer maybe!